### DISCUSSION TITLE
Setting up Stream Webserver using AnalogAudioStream as input

### DISCUSSION THREAD

**Discussion by davidlutta** [pschatzmann/arduino-audio-tools#370] [Q&A] [answered]
Hello, 

I am very new to Arduino programming and audio recording but I found this wonderful library yesterday which looks like it could help in my project. 

I am using a MAX9814 microphone module hooked up to an ESP32. The pins are setup as:

| MAX9814 |  ESP32 |
|------------|---------|
| 3.3v |  3.3v |
| gnd | gnd |
| out | GPIO 33|

I have modified one of the examples to look like this
```
#include "Arduino.h"
#include "BluetoothA2DPSource.h"
#include "AudioTools.h"
#include "AudioCodecs/CodecWAV.h"

AudioWAVServer server("fg_pi","password"); 
AnalogAudioStream in_stream;
uint16_t sample_rate = 44100;
uint8_t channels = 1; 
void setup() {
  // put your setup code here, to run once:
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Info);

  Serial.println("starting I2S-ADC...");
  in_stream.begin(in_stream.defaultConfig(RX_MODE));
  auto cfgRx = in_stream.defaultConfig(RX_MODE);
  cfgRx.sample_rate = sample_rate;
  cfgRx.channels = channels;
  in_stream.begin(cfgRx);
  Serial.println("ADC Started");

  server.begin(in_stream, cfgRx);
}

void loop() {
  server.copy();
}
```
Now when I run this this is the output I get

```
[I] AnalogAudioESP32.h : 68 - I2S_MODE_ADC_BUILT_IN
[I] AnalogAudioESP32.h : 196 - bool audio_tools::AnalogAudioStream::begin(audio_tools::AnalogConfig)
[I] AudioTypes.h : 53 - sample_rate: 44100
[I] AudioTypes.h : 54 - channels: 2
[I] AudioTypes.h : 55 - bits_per_sample: 16
[I] AnalogAudioESP32.h : 94 - input pin1: 34
[I] AnalogAudioESP32.h : 96 - input pin2: 0
[I] AnalogAudioESP32.h : 235 - RX_MODE
[I] AnalogAudioESP32.h : 68 - I2S_MODE_ADC_BUILT_IN
[I] AnalogAudioESP32.h : 196 - bool audio_tools::AnalogAudioStream::begin(audio_tools::AnalogConfig)
[I] AudioTypes.h : 53 - sample_rate: 44100
[I] AudioTypes.h : 54 - channels: 1
[I] AudioTypes.h : 55 - bits_per_sample: 16
[I] AnalogAudioESP32.h : 94 - input pin1: 34
```

After printing that on the monitor it then just prints out doted lines 
I am not even able to get the ip Address of the ESP module.

### What I am trying to do ?
I feel it is important to explain what I am trying to achieve, I would like to set up a remote recorder(microphone) that streams the audio output to my raspberry pi. 
I have converted my raspberry pi into a router and I am trying to make the ESP module stream it's audio to the raspberry pi.

Any help will be highly appreciated.

Thanks!

**Comment by pschatzmann** [comment, marked as answer]
I am afraid that this is not the right place to solve your infrastructure problems and I suggest that you take a step back. First try to connect to Wifi from an ESP32 w/o my framework.

There are plenty of tutorials and Google is your friend: If you can't connect, it is usually because of a wrong ssid or password. Or your Wifi Access point is requiring a security protocol that the ESP32 can't support.

Only after you succeed you can move to the next step!

There are plenty of alternative possibilities to reach your goal:
- The built-in simple Webserver is one of them, but you could also
- use the extended multiuser Webserver
- send out audio data via UDP or TCP/IP
- use a RTSPServer

But for all of them it is a precondition that the wifi is working properly!

**Comment by adriaanph** [comment]
@davidlutta i have almost the identical set-up as you. once you have managed to sort out your wifi issue, there are a few issues with your example code.

here's a slightly modified version of your code that should work:

```
#include "AudioTools.h"

AudioWAVServer server("ssid", "password"); // TODO: Change to match your WiFi router
AnalogAudioStream in;
ConverterAutoCenter<int16_t> converter; // Unsigned integer to +/-


void setup(){
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Info); // TODO: Change to ::Error once you've got it all sorted

  Serial.println("Starting input stream ...");
  auto config = in.defaultConfig(RX_MODE);
  config.sample_rate = 1600; // TODO: Change if you wish
  config.channels = 1;
  // RX automatically uses port 0 with pin GPIO34, but let's be verbose
  config.setInputPin1(GPIO_NUM_33); // Default 34. TODO: Change to match your wiring
  in.begin(config);
  Serial.println("Input stream started.");
  
  server.begin(in, config, &converter);
}


void loop() {
  server.copy();  
}
```

**Reply by davidlutta** [reply]
Thanks :) will give it another try
